src/storage/redis_metadata.h

```diff
@@ -97,22 +97,45 @@ class InternalKey {
   bool slot_id_encoded_;
 };
 
+constexpr uint8_t METADATA_64BIT_ENCODING_MASK = 0x80;
+constexpr uint8_t METADATA_TYPE_MASK = 0x0f;
+
 class Metadata {
  public:
+  // metadata flags
+  // <(1-bit) 64bit-common-field-indicator> 0 0 0 <(4-bit) redis-type>
+  // 64bit-common-field-indicator: make `expire` and `size` 64bit instead of 32bit
+  // NOTE: `expire` is stored in milliseconds for 64bit, seconds for 32bit
+  // redis-type: RedisType for the key-value
   uint8_t flags;
-  int expire;
+
+  // expire timestamp, in milliseconds
+  uint64_t expire;
+
+  // the current version: 53bit timestamp + 11bit counter
   uint64_t version;
-  uint32_t size;
 
- public:
-  explicit Metadata(RedisType type, bool generate_version = true);
+  // element size of the key-value
+  uint64_t size;
+
+  explicit Metadata(RedisType type, bool generate_version = true, bool use_64bit_common_field = true);
```

Comment:
But I am thinking if we set the parameter to `false` to allow users rollback to previous kvrocks version (i.e. users cannot rollback when they use the new encoding to write some data). Do you have some idea?

Reply:
Yes, I also had the same idea. 
So, maybe we can set 64-bit encoding in the constructor by default, and it will be overwritten by the Decode method in case of decoding from the 'old' format?

Reply:
Yes, you are right.

Reply:
I just thought that the `use_64bit_common_field` can be useful in unit tests to test backward compatibility, for example.

Reply:
> I just thought that the use_64bit_common_field can be useful in unit tests to test backward compatibility, for example.

Good idea.